### PR TITLE
Unpack postgres arrays for process idle times correctly

### PR DESCRIPTION
**Pull request by tomhughes**
The previous code just errored calling `Scan` on the row as the arrays are returned as strings that need to be decoded with `pq.Array`.

### CODE REVIEW

**Comment by tomhughes** [comment]
Add a second commit to fix the type of the total seconds, which should be a float.

File: collector/pg_process_idle.go [hunk 1]
```diff
@@ -111,7 +124,7 @@ func (PGProcessIdleCollector) Update(ctx context.Context, instance *instance, ch
 	}
 	secondsSumMetric := 0.0
 	if secondsSum.Valid {
-		secondsSumMetric = float64(secondsSum.Int64)
+		secondsSumMetric = float64(secondsSum.Float64)
```
Comment (SuperQ) [inline collector/pg_process_idle.go:127]: Simplification?

```suggestion
		secondsSumMetric = secondsSum.Float64
```

**Review by SuperQ — COMMENTED** [review]
Minor nit

File: collector/pg_process_idle.go (hunk 1 above)
Comment (tomhughes) [reply collector/pg_process_idle.go:127]: I don't know go at all, never mind the library that is using but if `Float64` is just a normal float that doesn't need a cast then sure.

Comment (tomhughes) [reply collector/pg_process_idle.go]: Seems to work so I've made the change.

**Review by sysadmind — APPROVED** [review]
I think we could just change the var to a pq.Array instead of manually scanning afterwards, but I think that this is better than nothing.

**Comment by tomhughes** [comment]
Oh I didn't realise that was possible... I was basing it on what the old queries based code was doing!

**Comment by SuperQ** [comment]
@tomhughes Want to make any changes or should we merge this as-is?

**Comment by tomhughes** [comment]
Sorry, was out last night so wasn't able to look at that suggestion... I've made the change to bind the arrays directly now.

**Comment by SuperQ** [comment]
Nice

**Pull request merged**